src/ssh_via_subprocess.py

```diff
@@ -0,0 +1,81 @@
+import subprocess
+import json
+import asyncio
+from shlex import join, quote
+
+
+async def run_remote_commands_for_data(username, password, host, url_to_curl, port='22'):
+    command = "sshpass -p {} ssh -p {} -o StrictHostKeyChecking=no {}@{} 'curl -s {}'".format(quote(password), quote(str(port)), quote(username), quote(host), quote(url_to_curl))
+    # string = username + '@' + host
+    # #command_1 = 'curl -s ' + url_to_curl
+    # command = ["sshpass", "-p", password, "ssh", "-p", port, "-o", "StrictHostKeyChecking=no", string, '"curl', '-s', url_to_curl+'"']
+    proc  = await asyncio.create_subprocess_shell(
+        command,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        shell=True
+    )
+    output, error = await proc.communicate()
+    try:
+        res = json.loads(output)
+        return {"success":res}
+    except Exception as e:
+        print("This is error while parsing the output!")
+        print(host," ",e, " ", error)
+        return {"error": "We have some error in running some commands into your server, please check your creds once, or try checking netdata installation!"}
+    else:
+        print("This is error while executing!")
+        print(host, " " ,error)
+        return {"error": "Unable to execute this operation!"}
+
+async def check_valid_ssh_and_netdata(username, password, host, port='22'):
+    url_to_curl = 'http://localhost:19999/api/v1/info'
+    command = "sshpass -p {} ssh -p {} -o StrictHostKeyChecking=no {}@{} 'curl -s {}'".format(quote(password), quote(str(port)), quote(username), quote(host), quote(url_to_curl))
```

Comment:
Same formatting command here also

Reply:
have used the quote function for shell escaping.
Imported the same, according to docs.
https://docs.python.org/3/library/shlex.html